### PR TITLE
Replace StringUtils.compare with inline code

### LINKED ISSUE #7376: Crash on v9.2-v9.5rc1: NoSuchMethodError in SiteSettingsTimezoneDialog
```
Fatal Exception: java.lang.NoSuchMethodError: No static method compare(Ljava/lang/String;Ljava/lang/String;)I in class Lorg/apache/commons/lang3/StringUtils; or its super classes (declaration of 'org.apache.commons.lang3.StringUtils' appears in /system/framework/framework.jar:classes2.dex)
       at org.wordpress.android.ui.prefs.SiteSettingsTimezoneDialog$6.compare(SiteSettingsTimezoneDialog.java:188)
       at org.wordpress.android.ui.prefs.SiteSettingsTimezoneDialog$6.compare(SiteSettingsTimezoneDialog.java:186)
       at java.util.TimSort.countRunAndMakeAscending(TimSort.java:320)
       at java.util.TimSort.sort(TimSort.java:199)
       at java.util.Arrays.sort(Arrays.java:1998)
       at java.util.Collections.sort(Collections.java:1900)
       at org.wordpress.android.ui.prefs.SiteSettingsTimezoneDialog.loadTimezones(SiteSettingsTimezoneDialog.java:186)
       at org.wordpress.android.ui.prefs.SiteSettingsTimezoneDialog.access$500(SiteSettingsTimezoneDialog.java:51)
       at org.wordpress.android.ui.prefs.SiteSettingsTimezoneDialog$3.onResponse(SiteSettingsTimezoneDialog.java:142)
       at org.wordpress.android.ui.prefs.SiteSettingsTimezoneDialog$3.onResponse(SiteSettingsTimezoneDialog.java:135)
       at com.android.volley.toolbox.StringRequest.deliverResponse(StringRequest.java:60)
       at com.android.volley.toolbox.StringRequest.deliverResponse(StringRequest.java:30)
       at com.android.volley.ExecutorDelivery$ResponseDeliveryRunnable.run(ExecutorDelivery.java:99)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5442)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:738)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:628)
```

Crashlytics ref: 5a5f27af8cb3c2fa6300a724

### PR DESCRIPTION
**Pull request by hypest**
Fixes #7376 

To test:
No steps available. That said, the crash seems to happen predominantly on Xiaomi devices running Android below v7. So, if you happen to have such a device, try to go to Mysite > Site settings > Timezone and see if the app crashes or not.

### CODE REVIEW

**Comment by nbradbury** [comment]
@hypest Fix looks good, but what do you think about moving the compare routine to our own `StringUtils` class?

**Comment by hypest** [comment]
Hmm, interesting proposition @nbradbury . I was trying to make this code "non-reusable", to favor the "default" of using the Apache lib provided one but, I guess we can move this to its own routine with some proper commenting. Will do that!

In the meantime, wdyt about replacing the use of `StringUtils.compareIgnoreCase()` in this same PR even though the crash is not related to it?

**Comment by hypest** [comment]
OK, I've carved out `compare` in its own method with c317f48 and `compareIgnoreCase` with 5bf33cb. Let me know and I can fork the second into its own PR if you prefer @nbradbury . Thanks!

**Comment by hypest** [comment]
Turns out the missing `compareIgnoreCase` case does occur in production. Opened #7396 for it. So, this PR should cover both tickets.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTimezoneDialog.java [hunk 1]
```diff
@@ -185,6 +185,8 @@ private void loadTimezones(@NonNull String responseJson) {
             // sort by label
             Collections.sort(timezones, new Comparator<Timezone>() {
                 public int compare(Timezone t1, Timezone t2) {
+                    // manually perform the StringUtils.compare() function. Some devices seem to be missing it.
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTimezoneDialog.java]: Looks like we can drop this comment now.

Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTimezoneDialog.java]: Done with b1b45dc!

**Comment by hypest** [comment]
Ready for another pass @nbradbury , thanks!

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**